counterpartylib/lib/transaction.py

```diff
@@ -362,24 +395,21 @@ def construct (db, tx_info, encoding='auto',
             # Pay‐to‐PubKeyHash, e.g.
             data_value = regular_dust_size
         data_output = (data_array, data_value)
+
+        if not dust_return_pubkey:
+            if encoding == 'multisig':
+                dust_return_pubkey = get_dust_return_pubkey(source, provided_pubkeys, encoding)
+            else:
+                dust_return_pubkey = None
     else:
+        data_array = []
         data_output = None
-    data_btc_out = sum([data_value for data_chunk in data_array])
+        dust_return_pubkey = None
 
+    data_btc_out = sum([data_value for data_chunk in data_array])
 
     '''Inputs'''
 
-    # Source.
-        # If public key is necessary for construction of (unsigned)
-        # transaction, use the public key provided, or find it from the
-        # blockchain.
-    if source:
-        script.validate(source)
-    if encoding == 'multisig':
```

Comment:
not super familiar with this part of the code so you'll have to excuse my ignorance:

is there any reason you moved this up under `if data:`? How would this change impact construction of something like a testnet burn tx (which has no `data`) using multisig encoding?